### ISSUE TITLE
interface conversion: interface is nil, not types.BaseVirtualDevice

### ISSUE THREAD

**Issue by lmayorga1980** [hashicorp/terraform-provider-vsphere#1201]
### Terraform Version

`0.12.26`

### vSphere Provider Version
 `v1.23.0`

### Errror

```
terraform plan --var-file vsphere.tfvars
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.vsphere_datacenter.dc: Refreshing state...
data.vsphere_network.network: Refreshing state...
data.vsphere_datastore.datastore: Refreshing state...
data.vsphere_resource_pool.pool: Refreshing state...
data.vsphere_virtual_machine.template: Refreshing state...

Error: rpc error: code = Unavailable desc = transport is closing


panic: interface conversion: interface is nil, not types.BaseVirtualDevice
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: 
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: goroutine 43 [running]:
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: github.com/hashicorp/terraform-provider-vsphere/vsphere/internal/virtualdevice.ReadSCSIBusType(0xc000180200, 0xb, 0x10, 0x1, 0xc0007aa7e8, 0x1)
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vsphere/vsphere/internal/virtualdevice/virtual_machine_device_subresource.go:707 +0x21d
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: github.com/hashicorp/terraform-provider-vsphere/vsphere.dataSourceVSphereVirtualMachineRead(0xc000394150, 0x204e8e0, 0xc0002584c0, 0xc000394150, 0x0)
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vsphere/vsphere/data_source_vsphere_virtual_machine.go:143 +0x696
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).ReadDataApply(0xc000218c00, 0xc0005d44c0, 0x204e8e0, 0xc0002584c0, 0xc0001d2eb8, 0x1, 0x0)
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vsphere/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:398 +0x88
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).ReadDataApply(0xc000218d80, 0xc0005e19a0, 0xc0005d44c0, 0xc0005d44c0, 0x0, 0x0)
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vsphere/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:451 +0x8f
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadDataSource(0xc000096508, 0x270d960, 0xc000518e70, 0xc00027b340, 0xc000096508, 0xc000518e70, 0xc0005eaa48)
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vsphere/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:1036 +0x45d
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadDataSource_Handler(0x2312b40, 0xc000096508, 0x270d960, 0xc000518e70, 0xc000200f00, 0x0, 0x270d960, 0xc000518e70, 0xc00057ad80, 0x43e)
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vsphere/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3225 +0x217
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000498160, 0x271a100, 0xc0005c4600, 0xc000260600, 0xc000350390, 0x32d5f30, 0x0, 0x0, 0x0)
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vsphere/vendor/google.golang.org/grpc/server.go:995 +0x460
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: google.golang.org/grpc.(*Server).handleStream(0xc000498160, 0x271a100, 0xc0005c4600, 0xc000260600, 0x0)
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vsphere/vendor/google.golang.org/grpc/server.go:1275 +0xd3d
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc00008c030, 0xc000498160, 0x271a100, 0xc0005c4600, 0xc000260600)
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vsphere/vendor/google.golang.org/grpc/server.go:710 +0xa1
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-09-02T14:06:47.060-0400 [DEBUG] plugin.terraform-provider-vsphere_v1.23.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vsphere/vendor/google.golang.org/grpc/server.go:708 +0xa1
2020/09/02 14:06:47 [ERROR] <root>: eval: *terraform.EvalReadData, err: rpc error: code = Unavailable desc = transport is closing
2020/09/02 14:06:47 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/09/02 14:06:47 [TRACE] [walkRefresh] Exiting eval tree: data.vsphere_virtual_machine.template
2020/09/02 14:06:47 [TRACE] vertex "data.vsphere_virtual_machine.template": visit complete
2020/09/02 14:06:47 [TRACE] vertex "data.vsphere_virtual_machine.template": dynamic subgraph encountered errors
2020/09/02 14:06:47 [TRACE] vertex "data.vsphere_virtual_machine.template": visit complete
2020-09-02T14:06:47.062-0400 [DEBUG] plugin: plugin process exited: path=/Users/user/repos/delphix-engine/.terraform/plugins/darwin_amd64/terraform-provider-vsphere_v1.23.0_x4 pid=8787 error="exit status 2"
2020/09/02 14:06:47 [TRACE] dag/walk: upstream of "provider.vsphere (close)" errored, so skipping
2020/09/02 14:06:47 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020-09-02T14:06:47.279-0400 [DEBUG] plugin: plugin exited



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.log" file that was created may contain 
sensitive information that must be redacted before it is safe to share 
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

from `crash.log`
```
2020/09/02 14:06:47 [ERROR] <root>: eval: *terraform.EvalReadData, err: rpc error: code = Unavailable desc = transport is closing
2020/09/02 14:06:47 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/09/02 14:06:47 [TRACE] [walkRefresh] Exiting eval tree: data.vsphere_virtual_machine.template
2020/09/02 14:06:47 [TRACE] vertex "data.vsphere_virtual_machine.template": visit complete
2020/09/02 14:06:47 [TRACE] vertex "data.vsphere_virtual_machine.template": dynamic subgraph encountered errors
2020/09/02 14:06:47 [TRACE] vertex "data.vsphere_virtual_machine.template": visit complete
2020-09-02T14:06:47.062-0400 [DEBUG] plugin: plugin process exited: path=/Users/luisma/repos/delphix-engine/.terraform/plugins/darwin_amd64/terraform-provider-vsphere_v1.23.0_x4 pid=8787 error="exit status 2"
2020/09/02 14:06:47 [TRACE] dag/walk: upstream of "provider.vsphere (close)" errored, so skipping
2020/09/02 14:06:47 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020-09-02T14:06:47.279-0400 [DEBUG] plugin: plugin exited
```

**Comment by bill-rich** [comment]
Fixed in #1188

**Comment by lmayorga1980** [comment]
@bill-rich can you provide some insight

```

provider "vsphere" {
  user                 = var.vsphere_user
  password             = var.vsphere_password
  vsphere_server       = var.vsphere_server
  allow_unverified_ssl = true
}

data "vsphere_datacenter" "dc" {
  name = "lab01"
}

data "vsphere_datastore" "datastore" {
  name          = "datastore1"
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_resource_pool" "pool" {
  name          = "cllab01/Resources"
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_network" "network" {
  name          = "VM-123"
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_virtual_machine" "template" {
  name          = "vmware-centos82"
  datacenter_id = data.vsphere_datacenter.dc.id
}
```

It is breaking on the template `resource` and I have never created a virtual machine per-se. This is supposed to read the existing template metadata.

**Comment by lmayorga1980** [comment]
Fixed on `1.24`
